### DISCUSSION TITLE
Requesting Knowledge Transfer Assistance for Student onboarding

### DISCUSSION THREAD

**Discussion by YujithIsura** [avinyafoundation/avinya_apps#264] [Announcements]
## Knowledge Transfer Request:
Hi @samisa-abeysinghe,
hope you're doing well. As you're aware, your expertise in student onboarding to the Avinya system using spreadsheets has been needed to onboard our second batch of students to the Avinya system.
Given your expertise in the student onboarding process and the efficient use of spreadsheets, we kindly request your assistance in transferring this knowledge to our team. Your insights into the following areas would be highly beneficial:

* The steps involved in the student onboarding process.
* How to accurately collect and input student data into spreadsheets.
* Any tips, best practices, or potential challenges you've encountered.

### Contribution Method:
If you're open to helping us with this knowledge transfer, there are a few ways you could contribute:

* A brief written guide or documentation outlining the process step by step.
* Participating in a virtual meeting where you could demonstrate and explain the process to us.

Thank you for your assistance with this matter.

**Comment by samisa-abeysinghe** [comment]
1. The steps involved in the onboarding process is already doccumented in the sequnce diagram that was shared sometibe back 
2. The simplest model is to follow the same format as "Onboariding V3" sheet used for the last batch. There are tabs that extract from the mater sheet and compute the email address (Digital ID) and the like. You need to be able to map the data in the seet into person table colums in addition to computing the digital ID
3. When adding the students to the Asgardeo systsem and the digital platforms like Google Workspaces, I used APIs and Python scrips that I have already shared with you. These cannot be shared in public domain as there are sensitve information like keys and names in there. The defining criteria is the digital ID. It is almost like a UUID for the whole of the system, including mapping the ID acorss the production DB, Asgardeo and systems like Google workspace. 
4. The best form of documentation is the Onboarding sheet V3 used for the first batch as it has all the formulas and the data format. Writing this on another doc is redunfdant as you will have to map that to the new sheet asformular anyway. Please look into the model used by the attendance sheet and the grading sheet how the digital ID from Onboarding sheet is used to map to the other data recording sheets. Then when you map that in tern to the DB design driven by the digital ID mapping, the design of which you are already familar with, you will be able to get the mapping across the app stack and sheets easily. 
5. If you need further details, I can get on a call.

**Reply by YujithIsura** [reply]
Thank you so much @samisa-abeysinghe will you be available sometime today to get on a call?
